File: android/app/src/main/java/com/example/app/domain/navcaddy/models/ConfidenceThresholds.kt
package caddypro.domain.navcaddy.models

/**
 * Confidence thresholds for intent routing decisions.
 *
 * These thresholds determine whether to route directly, confirm with user,
 * or request clarification.
 *
 * Spec reference: navcaddy-engine.md Q2 (RESOLVED)
 */
object ConfidenceThresholds {
    /**
     * Minimum confidence to route directly to a destination.
     * Confidence >= 0.75 routes immediately without confirmation.
     */
    const val ROUTE_THRESHOLD = 0.75f

    /**
     * Minimum confidence to confirm intent before routing.
     * Confidence in range [0.50, 0.75) prompts user confirmation.
     */
    const val CONFIRM_THRESHOLD = 0.50f

    /**
     * Below this threshold, request clarification.
     * Confidence < 0.50 triggers clarification with suggested intents.
     */
    const val CLARIFY_THRESHOLD = 0.50f

    /**
     * Determines the appropriate action based on confidence level.
     *
     * @param confidence The intent classification confidence (0-1)
     * @return The threshold action to take
     */
    fun actionForConfidence(confidence: Float): ThresholdAction = when {
        confidence >= ROUTE_THRESHOLD -> ThresholdAction.ROUTE
        confidence >= CONFIRM_THRESHOLD -> ThresholdAction.CONFIRM
        else -> ThresholdAction.CLARIFY
    }
}

/**
 * Actions that can be taken based on confidence thresholds.
 */
enum class ThresholdAction {
    /** Navigate directly to the destination without confirmation */
    ROUTE,
    /** Ask user to confirm the detected intent before navigating */
    CONFIRM,
    /** Request clarification with suggested intents */
    CLARIFY
}

File: ios/App/Sources/Core/Domain/NavCaddy/ConfidenceThresholds.swift
import Foundation

/// Confidence threshold constants for intent routing decisions.
///
/// Spec Q2 (Resolved): Balanced thresholds approach
/// - Route: >= 0.75 (high confidence, direct navigation)
/// - Confirm: 0.50 - 0.74 (medium confidence, ask for confirmation)
/// - Clarify: < 0.50 (low confidence, request clarification)
enum ConfidenceThresholds {
    /// Minimum confidence to route directly without confirmation
    static let route: Double = 0.75

    /// Minimum confidence to show confirmation before routing
    static let confirm: Double = 0.50

    /// Below this threshold, clarification is required
    /// (Anything below confirm threshold triggers clarification)
    static let clarify: Double = 0.50

    /// Determines the action to take based on confidence score
    static func action(for confidence: Double) -> ThresholdAction {
        if confidence >= route {
            return .route
        } else if confidence >= confirm {
            return .confirm
        } else {
            return .clarify
        }
    }
}

/// Represents the action to take based on confidence threshold
enum ThresholdAction {
    case route      // High confidence: navigate directly
    case confirm    // Medium confidence: ask for confirmation
    case clarify    // Low confidence: request clarification
}
